reference/zrc6.scilla

```diff
@@ -0,0 +1,980 @@
+transition AddOperator(to: ByStr20)
+  RequireNotSelf to _sender;
+  
+  opt_bal <- balances[_sender];
+  balance = get_bal opt_bal;
+  
+  is_balance_zero = builtin eq zero balance;
+  (* _sender should have at least 1 token *)
+  match is_balance_zero with 
+  | True =>    
+    err = NotTokenOwnerError;
+    ThrowError err
+  | False =>
+    has_operator <- exists operators[_sender][to];
+    match has_operator with
+    | False =>
+      (* Add operator *)
+      operators[_sender][to] := verdad
+    | True =>
+      err = OperatorFoundError;
+      ThrowError err
+    end;
+    e = {
+      _eventname : "AddOperator";
+      initiator : _sender;
+      to : to
+    };
+    event e;
+    msg_to_sender = {
+      _tag : "ZRC6_AddOperatorCallback";
+      _recipient : _sender;
+      _amount : Uint128 0;
+      to : to
+    };
+    msgs = one_msg msg_to_sender;
+    send msgs
+  end
+end
+
+(* Removes `to` from operator for the `_sender`. *)
+(* @Requirements:  *)
+(* - `_sender` should be the token owner. Otherwise, it should throw `NotTokenOwnerError` *)
+(* - `_sender` should not be `to`. Otherwise, it should throw `SelfError` *)
+(* - `to` should be already an operator. Otherwise, it should throw `OperatorNotFoundError` *)
+transition RemoveOperator(to: ByStr20)
+  RequireNotSelf to _sender;
+  has_operator <- exists operators[_sender][to];
+  match has_operator with
+  | False =>
+    err = OperatorNotFoundError;
+    ThrowError err
+  | True =>
+    (* Remove operator *)
+    delete operators[_sender][to]
+  end;
+  e = {
+    _eventname : "RemoveOperator";
+    initiator : _sender;
+    to : to
+  };
+  event e;
+  msg_to_sender = { 
+    _tag : "ZRC6_RemoveOperatorCallback";
+    _recipient : _sender;
+    _amount : Uint128 0;
+    to : to
+  };
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* Transfers `token_id` from the token owner to `to`.  *)
+(* @Requirements: *)
+(* - The contract should not be paused. Otherwise, it should throw `PausedError` *)
+(* - `to` should not be the zero address. Otherwise, it should throw `ZeroAddressDestinationError` *)
+(* - `to` should not be `_this_address`. Otherwise, it should throw `ThisAddressDestinationError` *)
+(* - `token_id` should exist. Otherwise, it should throw `TokenNotFoundError` *)
+(* - `_sender` should be a token owner, spender, or operator. Otherwise, it should throw `NotAllowedToTransferError` *)
+(* - `_sender` should not be `to`. Otherwise, it should throw `SelfError` *)
+transition TransferFrom(to: ByStr20, token_id: Uint256)
+  RequireNotPaused;
+  RequireValidDestination to;
+
+  opt_token_owner <- token_owners[token_id];
+  match opt_token_owner with
+  | None =>
+    err = TokenNotFoundError;
+    ThrowError err
+  | Some token_owner =>
+    RequireAccessToTransfer token_owner token_id;
+    RequireNotSelf token_owner to;
+    
+    (* change token_owner for that token_id *)
+    token_owners[token_id] := to;
+
+    delete spenders[token_id];
```

Comment:
I think we should also delete the operator

Reply:
A token owner can have multiple operators and `operator` is not bound to a single token unlike `spender`

`field operators: Map ByStr20 (Map ByStr20 Dummy) = Emp ByStr20 (Map ByStr20 Dummy)`

Operators can
- transfer a token
- burn a token
- add/remove a spender of a token

Reply:
Got it. One is bounded to token, while the other is bound to an addrress